### PR TITLE
[Next] add an experimental command `az next` 

### PR DESCRIPTION
**Pull request by zhoxing-ms**
This pr add an experimental command az next which can recommend next command for you.

# Architecture Overview

There are mainly 3 classes in scheduled task:

## Kusto Client

You can use `get_data` function to get data from Kusto database. Please note that the Kusto query string hard codes the time period. If you need to change the function's schedule, you also need to change the time period in the query string.

### Returned Value

Returned value is a dictionary whose key is `uid` and value is a list of tuples which contain `command` `param` `os_type` `source` `begin_time` `end_time`. 

## Cosmos Client

You can use the Cosmos client to query data and insert items to Cosmos DB. We query all items in Cosmos DB to restore previous state of the `CommandTable`. And then use data from Kusto client to update the `CommandTable`. Finally, write back new data to Cosmos DB.

### Returned Value

Query return an iterable dict. You can find the API reference [here](https://docs.microsoft.com/en-us/python/api/azure-cosmos/azure.cosmos.containerproxy?view=azure-python#query-items-query--parameters-none--partition-key-none--enable-cross-partition-query-none--max-item-count-none--enable-scan-in-query-none--populate-query-metrics-none----kwargs-).

## Command Table

This table is used for calculating the next command for each command.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
az next

File: src/next/setup.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    'Programming Language :: Python :: 3.6',
+    'Programming Language :: Python :: 3.7',
+    'Programming Language :: Python :: 3.8',
+    'License :: OSI Approved :: MIT License',
+]
+
+# TODO: Add any additional SDK dependencies here
+DEPENDENCIES = []
+
+with open('README.rst', 'r', encoding='utf-8') as f:
```
Comment (jsntcy) [inline src/next/setup.py:38]: Remove either README.rst or README.md, just keep one.

File: src/next/azext_next/azext_metadata.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+    "azext.isPreview": true,
+    "azext.minCliCoreVersion": "2.15.0"
```
Comment (jsntcy) [inline src/next/azext_next/azext_metadata.json:3]: 2.19.0?

File: src/next/azext_next/azext_metadata.json [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{
+    "azext.isPreview": true,
```
Comment (jsntcy) [inline src/next/azext_next/azext_metadata.json:2]: "azext.isExperimental": true

File: src/next/azext_next/_client_factory.py [hunk 4]
```diff
@@ -0,0 +1,8 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+
+def cf_next(cli_ctx, *_):  # pylint: disable=unused-argument
```
Comment (jsntcy) [inline src/next/azext_next/_client_factory.py:7]: Do we need this factory as it just returns None?

File: src/next/azext_next/commands.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# pylint: disable=line-too-long
+# pylint: disable=unused-import
+from azure.cli.core.commands import CliCommandType
+from azext_next._client_factory import cf_next
```
Comment (jsntcy) [inline src/next/azext_next/commands.py:9]: Useless import?

**Review by jsntcy — APPROVED** [review]
:shipit:

File: src/next/setup.py (hunk 1 above)
Comment (zhoxing-ms) [reply src/next/setup.py]: done

File: src/next/azext_next/azext_metadata.json (hunk 2 above)
Comment (zhoxing-ms) [reply src/next/azext_next/azext_metadata.json]: done

File: src/next/azext_next/azext_metadata.json (hunk 3 above)
Comment (zhoxing-ms) [reply src/next/azext_next/azext_metadata.json]: done

File: src/next/azext_next/_client_factory.py (hunk 4 above)
Comment (zhoxing-ms) [reply src/next/azext_next/_client_factory.py]: done

File: src/next/azext_next/commands.py (hunk 5 above)
Comment (zhoxing-ms) [reply src/next/azext_next/commands.py]: Yes, I have deleted it

**Pull request merged**